armory/scenarios/scenario.py

```diff
@@ -227,17 +227,22 @@ def load_metrics(self):
             computational_resource_dict=metrics_logger.computational_resource_dict,
         )
 
-        export_samples = self.config["scenario"].get("export_samples")
-        if export_samples is not None and export_samples > 0:
-            sample_exporter = SampleExporter(
-                self.scenario_output_dir, self.test_dataset.context, export_samples
-            )
+        self.metrics_logger = metrics_logger
+
+    def load_sample_exporter(self):
+        self.num_export_samples = self.config["scenario"].get("export_samples")
+        if self.num_export_samples is not None and self.num_export_samples > 0:
+            sample_exporter = self._load_sample_exporter()
         else:
             sample_exporter = None
-
-        self.metrics_logger = metrics_logger
         self.sample_exporter = sample_exporter
```

Comment:
Since `_load_sample_exporter()` is required, what if we always instantiate the sample_exporter, and add logic to `evaluate_current()` to determine when to call `self.export_samples()`? This would ensure that in interactive use the user can always call `export_samples()`.

Reply:
In 8f376a0, I made a change s.t. an Exporter is always loaded but the scenario attribute `self.num_export_batches` defaults to 0 (rather than `None`). Then in `evaluate_current()`, a simple check of ` if self.num_export_batches > self.sample_exporter.saved_batches` is done